### ISSUE TITLE
Upgrade client command failing with "failed while fetching from chain the upgraded client state"

### ISSUE THREAD

**Issue by V-Staykov** [informalsystems/ibc-rs#1632]
## Crate
`ibc-relayer` and `ibc-relayer-cli`

## Summary of Bug
When trying to upgrade a chain and using the command `hermes upgrade client` for the counter party chain I get the following error:
```
Error: 
   0: failed while trying to upgrade client id 07-tendermint-0 for chain <chain-id-b>: failed while fetching from chain the upgraded client state: Conversion from a protobuf `Any` into a domain type failed
   1: Conversion from a protobuf `Any` into a domain type failed
   2: 
   2:    0: error converting message type into domain type: 
   2:          0: the client state was not found
   2:       Location:
   2:          /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/flex-error-0.4.4/src/tracer_impl/eyre.rs:10

```
It's the same with `hermes upgrade clients`. ``hermes update client` seems to work fine. Also relaying tokens between the networks works fine.

Doesn't matter when I try to run the upgrade command, meaning before the halt, during or after.

## Version
v0.9.0

## Steps to Reproduce
1. Bring up 2 cosmos-sdk v0.44.3 and ibc-go v1.1.0 networks.
2. Setup the relayer between them and create a channel.
3. Run `hermes upgrade client <chain-id-a> <client-a>`

## Acceptance Criteria
Upgrade command running without this error.

____

## For Admin Use

- [x] Not duplicate issue
- [x] Appropriate labels applied
- [x] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by adizere** [comment]
Hi,
Thanks for opening this issue.

In order to upgrade any IBC client, the network which that client is targeting must also undergo an upgrade. In other words, you usually need to run `upgrade-chain` CLI before running `upgrade client`. The procedure described in the [testing client upgrade](https://hermes.informal.systems/commands/upgrade/test.html) section of the Hermes guide gives more details, in particular steps 3 and 4.

Let me know if this solves the issue you noticed.

**Comment by V-Staykov** [comment]
> Let me know if this solves the issue you noticed.
I tried with creating and voting a proposal from a node on the chain. Is it supposed to be only from the hermes CLI?
I didn't include it in the issue, because the error is the same no matter if there is a proposal, if the proposal has passed or the chain is already halted.

**Comment by V-Staykov** [comment]
Okay, after I tried your guide I found the following. This error seems to be thrown if the upgrade proposal has it's UpgradedClientState in tset to null. This was happening to me, because I was testing with an upgrade proposal from the chain nodes CLI and wasn't including this state.

**Comment by V-Staykov** [comment]
Okay - new problem. Seems the UpgradedClientState in the upgrade plan was depricated in Cosmos sdk v0.43. The tutorial you have is with gaia v4.2.1, which runs on pre Cosmos sdk v0.42 even. When I am running a chain on Cosmos sdk v0.44+ I am unable to even create a proposal with UpgradedClientState and follow their tutorial, neither yours, for upgrade. 

All in all the problem seems to be with braking changes to cosmos sdk version.

**Comment by adizere** [comment]
> When I am running a chain on Cosmos sdk v0.44+ I am unable to even create a proposal with UpgradedClientState and follow their tutorial, neither yours, for upgrade.

Ouch, this is bad, it might be a regression! I think in this PR https://github.com/informalsystems/ibc-rs/pull/1289 we introduced support for submitting `upgrade-chain` CLIs that are compatible with both 0.42 (using `--legacy` flag) and 0.43+ (by default). Can you please detail the bug so we can reproduce it and track down the problem?

We'll need chain SDK versions, Hermes version, and a basic recipe of commands you're using to set everything up and produce the problem. Many thanks in advance!

**Comment by V-Staykov** [comment]
Well I am setting up a chain on Cosmos SDK v0.44.3 and ibc-go v1.1.0. I first tried to go by [this](https://docs.cosmos.network/master/ibc/upgrades/quick-guide.html) guide, but the problem is, after Cosmos SDK v0.43 I can't create proposal with UpgradedClientState in it. If I create a normal proposal, then the UpgradeClient from Hermes CLI fails to parse the protobuf because the client state is missing.

If I try to create the upgrade proposal from the Hermes CLI, exactly as in your guide [here](https://hermes.informal.systems/commands/upgrade/test.html), but with my chains, it fails with something (from memory) about not having a handler for this.

P.S. Now I see that since v0.43 there is a separate upgrade tx for the IBC client, but I couldn't find any explanation on this. Do I need to create both gov UpgradeProposal and a client upgrade proposal? Also I guess you are doing this from the Hermes CLI, but I can't make it work.

Basically I guess the problem is with me, but I can't seem to find what exactly is needed in the current release.

**Comment by V-Staykov** [comment]
This was totally a problem on my side as I suspected. It turned out I didn't implement the  IBC-Go module and was missing some of the msg handlers.
